sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveCatalog.scala

```diff
@@ -85,6 +85,13 @@ private[spark] class HiveCatalog(client: HiveClient) extends ExternalCatalog wit
     withClient { getTable(db, table) }
   }
 
+  /**
+   * Switch our client to one that belongs to the new session.
+   */
+  def newSession(newClient: HiveClient): this.type = {
```

Comment:
i'm not sure about this one, but should we just consider creating the hive client in the ctor itself?